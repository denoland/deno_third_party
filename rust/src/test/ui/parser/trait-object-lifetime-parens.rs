// compile-flags: -Z parse-only -Z continue-parse-after-error

fn f<T: Copy + ('a)>() {} //~ ERROR parenthesized lifetime bounds are not supported

fn main() {
    let _: Box<Copy + ('a)>; //~ ERROR parenthesized lifetime bounds are not supported
    let _: Box<('a) + Copy>; //~ ERROR expected type, found `'a`
}
