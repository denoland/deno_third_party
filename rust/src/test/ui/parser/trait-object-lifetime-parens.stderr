error: parenthesized lifetime bounds are not supported
  --> $DIR/trait-object-lifetime-parens.rs:3:19
   |
LL | fn f<T: Copy + ('a)>() {} //~ ERROR parenthesized lifetime bounds are not supported
   |                   ^

error: parenthesized lifetime bounds are not supported
  --> $DIR/trait-object-lifetime-parens.rs:6:26
   |
LL |     let _: Box<Copy + ('a)>; //~ ERROR parenthesized lifetime bounds are not supported
   |                          ^

error: expected type, found `'a`
  --> $DIR/trait-object-lifetime-parens.rs:7:17
   |
LL |     let _: Box<('a) + Copy>; //~ ERROR expected type, found `'a`
   |         -       ^^
   |         |
   |         while parsing the type for `_`

error: aborting due to 3 previous errors

